src/server/index.tsx

```diff
@@ -65,18 +67,34 @@ export const createServer = ({
   };
 
   const cdnURL = env.PUBLIC_URL || "'none'";
+  const connectSrc = [
+    // Relax the connect-src if using the proxy otherwise
+    // Use the env var or 'none' if the API host isn't set.
+    env.REACT_APP_USE_INSECURE_PROXY === 'true'
+      ? "'self'"
+      : env.REACT_APP_API_HOST || "'none'",
+  ];
+
+  if (env.REACT_APP_SENTRY_DSN) {
+    try {
+      const { host, protocol } = url.parse(env.REACT_APP_SENTRY_DSN);
+
+      if (host && protocol) {
+        connectSrc.push(`${protocol}//${host}`);
+      }
```

Comment:
This should have an `else` branch that throws an error so that the catch down below will handle it